File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/paymentproduct/validation/FixedList.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.paymentproduct.validation;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * POJO which holds the FixedList data.
 * Used for validation.
 */
public class FixedList implements Serializable {

	private static final long serialVersionUID = -7191166722186646029L;

	private List<String> allowedValues = new ArrayList<>();

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public FixedList() {}

	public List<String> getAllowedValues(){
		return allowedValues;
	}
}

File: onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/paymentproduct/validation/FixedList.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.paymentproduct.validation

import java.io.Serializable

/**
 * Data class which holds the FixedList data.
 * Used for validation.
 */
data class FixedList(
    val allowedValues: MutableList<String?>? = mutableListOf()
) : Serializable {

    companion object {
        @Suppress("Unused")
        private val serialVersionUID = -7191166722186646029L
    }
}
